src/Apps/ArtAdvisor/06-Discovery-Neartext/App.tsx

```diff
@@ -0,0 +1,148 @@
+import { FC, useEffect, useState } from "react"
+import { Box, Select, SkeletonBox, Spacer, Text } from "@artsy/palette"
+import {
+  DiscoveryArtwork,
+  DiscoveryUser,
+  UUID,
+} from "Apps/ArtAdvisor/06-Discovery-Neartext/types"
+import { Rail } from "Components/Rail/Rail"
+import _ from "lodash"
+import { Artwork } from "Apps/ArtAdvisor/06-Discovery-Neartext/components/Artwork"
+import { Concepts } from "Apps/ArtAdvisor/06-Discovery-Neartext/components/Concepts"
+
+export const App: FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [users, setUsers] = useState<DiscoveryUser[]>([])
+  const [userId, setUserId] = useState<UUID>()
+  const [user, setUser] = useState<DiscoveryUser>()
+  const [conceptText, setConceptText] = useState<string>("")
+  const [concepts, setConcepts] = useState<string[]>([])
+  const [artworks, setArtworks] = useState<DiscoveryArtwork[]>([])
+
+  useEffect(() => {
+    console.log("[Discovery] mount effect")
+
+    async function fetchUsers() {
+      const response = await fetch("/api/advisor/6/users")
+      const users = await response.json()
+      return users
+    }
+
+    fetchUsers().then(setUsers)
+  }, [])
+
+  useEffect(() => {
+    console.log("[Discovery] user effect for", userId)
```

Comment:
I handled the data-fetching side effects by having three distinct `useEffect`s:

- one that fetches the `DiscoveryUser` list on initial mount
- one that fetches a full `DiscoveryUser` record when a user is selected from the dropdown (this one)
- one that fetches `DiscoveryArtworks` whenever the concept list or user (likes/dislikes) is changed